packages/flutter/lib/src/material/switch.dart

```diff
@@ -385,6 +387,40 @@ class Switch extends StatelessWidget {
   /// the [Switch] track has no outline by default.
   final MaterialStateProperty<Color?>? trackOutlineColor;
 
+  /// {@template flutter.material.switch.trackOutlineWidth}
+  /// The outline width of this [Switch]'s track.
+  ///
+  /// Resolved in the following states:
+  ///  * [MaterialState.selected].
+  ///  * [MaterialState.hovered].
+  ///  * [MaterialState.focused].
+  ///  * [MaterialState.disabled].
+  ///
+  /// {@tool snippet}
+  /// This example resolves the [trackOutlineWidth] based on the current
+  /// [MaterialState] of the [Switch], providing a different outline width when it is
+  /// [MaterialState.disabled].
+  ///
+  /// ```dart
+  /// Switch(
+  ///   value: true,
+  ///   onChanged: (_) => true,
+  ///   trackOutlineWidth: MaterialStateProperty.resolveWith<double?>((Set<MaterialState> states) {
+  ///     if (states.contains(MaterialState.disabled)) {
+  ///       return 5.0;
+  ///     }
+  ///     return null; // Use the default width.
+  ///   }),
+  /// )
+  /// ```
+  /// {@end-tool}
+  /// {@endtemplate}
+  ///
+  /// In Material 3, the outline width defaults to 2.0 in the selected
+  /// state and [ColorScheme.outline] in the unselected state. In Material 2,
```

Comment:
What about the hovered and focused states?

Reply:
Ah thanks for catching this. Just updated. The outline width in M3 should be 2.0 in all states.